### ISSUE TITLE
Pinning exact dependency versions

### ISSUE THREAD

**Issue by omnisci3nce** [riot-ml/riot#74]
I just cloned Riot and tried to get set up on my macbook and noticed that the `tls` dependency has been updated to 0.17.4, and our `dune-project` specifies `(tls (>="0.17.3"))`. Unfortunately on 0.17.4 Riot does not build due to a slight interface change.

<details>

<summary>Exact error</summary>

```bash
File "riot/lib/ssl.ml", line 97, characters 11-47:
97 |       | Ok (state', `Response resp, `Data data) ->
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: This pattern matches values of type 'a * 'b * 'c
       but a pattern was expected which matches values of type
         Tls.Engine.state * [ `Eof ] option *
         [ `Response of IO.Cstruct.t option ] *
         [ `Data of IO.Cstruct.t option ]
```
</details>

This is more of a conversation starter as we've been seeing a few people running into build issues - is it worth starting to pin dependencies to exact versions a.k.a `=` instead of `>=` ?

If so I can create a PR that sets the current non-riot deps to `=`. Or you can since it should be such a small change 😂

**Comment by leostera** [comment]
Oh yes, this was pinned in https://github.com/ocaml/opam-repository/pull/25594/files#diff-a74b089612c7d97a1a1c4e3063303ca5af3eead0fd9e5e0589fb67178482d2b4R26 so we'll have to fix it to `0.17.3` for now.

**Comment by omnisci3nce** [comment]
Ah I see. So if I install `riot` via opam then it will work correctly presumably. I didn't know this was how the package versions are resolved using opam.

**Comment by leostera** [comment]
I've fixed the TLS issue by pinning to 0.17.3 in the repo too. Closing this for now, thanks for opening @omnisci3nce! ✨
